c7n/resources/route53.py

```diff
@@ -194,6 +194,39 @@ def process_resource_set(self, client, domains, keys):
                 TagsToDelete=keys)
 
 
+@HostedZone.action_registry.register('delete')
+class Delete(BaseAction):
+    """Action to delete Route 53 hosted zones
+
+    It is recommended to use a filter to avoid unwanted deletion of R53 hosted zones
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: route53-delete-testing-hosted-zones
+                resource: aws.hostedzone
+                filters:
+                  - 'tag:TestTag': present
+                actions:
+                  - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('route53:DeleteHostedZone')
+
+    def process(self, hosted_zones):
+        with self.executor_factory(max_workers=3) as w:
+            list(w.map(self.process_hosted_zones, hosted_zones))
+
+    def process_hosted_zones(self, hosted_zone):
+        client = local_session(self.manager.session_factory).client('route53')
```

Comment:
this will be resolved by removing the concurrency but generally we prefer to do client construction in main thread and hand it to worker threads. they are pretty slow to create.

Reply:
addressed